### PR TITLE
use new auth flow and fix switch account interaction

### PR DESCRIPTION
**Pull request by xavluiz**

### CODE REVIEW

**Review by brettmstevens7 — APPROVED** [review]

File: src/DataController.ts [hunk 1]
```diff
@@ -329,11 +316,9 @@ export async function writeProjectCommitDashboardByStartEnd(
     end,
     projectIds
 ) {
-    const qryStr = `?start=${start}&end=${end}&projectIds=${projectIds.join(
-        ","
-    )}`;
-    const api = `/projects/codeSummary${qryStr}`;
-    const result = await softwareGet(api, getItem("jwt"));
+    const qryStr = `?start=${start}&end=${end}`;
```
Comment (pbstriker38) [inline src/DataController.ts:319]: Why isn't this moving to the body as well?

File: src/Util.ts [hunk 2]
```diff
@@ -300,6 +300,14 @@ export function getItem(key) {
     return fileIt.getJsonValue(getSoftwareSessionFile(), key);
 }
 
+export function getPluginUuid() {
+    return fileIt.getJsonValue(getDeviceFile(), "plugin_uuid");
+}
+
+export function setPluginUuid(value: string) {
+    fileIt.setJsonValue(getDeviceFile(), "plugin_uuid", value);
```
Comment (pbstriker38) [inline src/Util.ts:308]: Should we add a check here to make sure a UUID doesn't already exist. So that we don't accidentally overwrite the UUID.

Comment (xavluiz) [reply src/Util.ts:309]: yeah, good point. i'll do that

File: src/DataController.ts (hunk 1 above)
Comment (xavluiz) [reply src/DataController.ts]: that was updated, thanks for the suggestion

**Pull request merged**